iotfunctions/anomaly.py

```diff
@@ -139,40 +143,54 @@ def execute(self, df):
+                         ', Overlap: ' + str(self.windowoverlap) + ', Inputsize: ' + str(temperature.size))
 
             if temperature.size <= self.windowsize:
-                logger.debug(str(temperature.size) + ' <= ' + str(self.windowsize))
-                df_copy.loc[[entity]] = 0.0001
+                logger.info(str(temperature.size) + ' <= ' + str(self.windowsize))
+                # df_copy.loc[[entity]] = 0.0001
             else:
-                logger.debug(str(temperature.size) + str(self.windowsize))
+                logger.debug('Size:' + str(temperature.size) + ', Windowsize: ' + str(self.windowsize) +
+                             ', Type: ' + str(temperature.dtype))
+                dfe[self.output_item] = 0.0007
                 # Fourier transform:
                 #   frequency, time, spectral density
-                frequency_temperature, time_series_temperature, spectral_density_temperature = signal.spectrogram(
-                    temperature, fs=self.frame_rate, window='hanning',
-                    nperseg=self.windowsize, noverlap=self.windowoverlap,
-                    detrend=False, scaling='spectrum')
-
-                # cut off freqencies too low to fit into the window
-                frequency_temperatureb = (frequency_temperature > 2/self.windowsize).astype(int)
-                frequency_temperature = frequency_temperature * frequency_temperatureb
-                frequency_temperature[frequency_temperature == 0] = 1 / self.windowsize
-
-                highfrequency_temperature = frequency_temperature.copy()
-                lowfrequency_temperature = frequency_temperature.copy()
-                highfrequency_temperature[highfrequency_temperature <= FrequencySplit] = 0
-                lowfrequency_temperature[lowfrequency_temperature > FrequencySplit] = 0
-
-                # Compute energy = frequency * spectral density over time in decibel
                 try:
-                    lowsignal_energy = np.log10(np.dot(spectral_density_temperature.T, lowfrequency_temperature))
-                    highsignal_energy = np.log10(np.dot(spectral_density_temperature.T, highfrequency_temperature))
+                    frequency_temperature, time_series_temperature, spectral_density_temperature = signal.spectrogram(
+                        temperature, fs=self.frame_rate, window='hanning',
+                        nperseg=self.windowsize, noverlap=self.windowoverlap,
+                        detrend=False, scaling='spectrum')
+
+                    # cut off freqencies too low to fit into the window
+                    frequency_temperatureb = (frequency_temperature > 2/self.windowsize).astype(int)
+                    frequency_temperature = frequency_temperature * frequency_temperatureb
+                    frequency_temperature[frequency_temperature == 0] = 1 / self.windowsize
+
+                    highfrequency_temperature = frequency_temperature.copy()
```

Comment:
the better naming convention would be high_frequency_temperature  instead of highfrequency_temperature.